src/language/ast.js

```diff
@@ -398,15 +403,15 @@ export type ObjectTypeDefinitionNode = {
   name: NameNode,
   interfaces?: ?Array<NamedTypeNode>,
   directives?: ?Array<DirectiveNode>,
-  fields: Array<FieldDefinitionNode>,
+  fields?: ?Array<FieldDefinitionNode>,
 };
 
 export type FieldDefinitionNode = {
   kind: 'FieldDefinition',
   loc?: Location,
   description?: ?StringValueNode,
   name: NameNode,
-  arguments: Array<InputValueDefinitionNode>,
+  arguments?: ?Array<InputValueDefinitionNode>,
```

Comment:
I'm not entirely clear on why this is now a nullable and optional argument. Is this the differences between:
```
some_field
```
vs
```
some_field()
```
vs
```
some_field(someArg: ArgType, someOtherArg: ArgType)
```
?

Basically, in this case, what's the difference between an unset `arguments` vs `arguments: []`?

Reply:
This was fixing a mistake to bring this into consistency with other places where an array of things is optional. 

`some_field()` is a syntax error, so we only care about the difference between `some_field` and `some_field(arg: Type)`.

This mirrors the same thing with field selection sets where `field {}` is a syntax error, and we care about the difference between `field` and `field { subfield }`.

In both of these cases, we allow the AST to provide an unset property as a short-hand convenience to mean the same thing as that property being defined with an empty array `[]`. 

Our parser always produces `[]` for these to ensure the same hidden class, but the flow types ensure any code that operates on the AST is resilient against the use of this convenience which makes it easier to build ASTs from other utilities and provides some flexibility in the future should we want to have our parser omit empty arrays if it ever becomes a memory burden